mobly/controllers/android_device_lib/snippet_client_v2.py

```diff
@@ -403,9 +403,10 @@ def make_connection(self):
 
   def _forward_device_port(self):
     """Forwards the device port to a host port."""
-    if not self.host_port:
-      self.host_port = utils.get_available_host_port()
-    self._adb.forward([f'tcp:{self.host_port}', f'tcp:{self.device_port}'])
+    host_port = self.host_port or 0
+    stdout = self._adb.forward([f'tcp:{host_port}', f'tcp:{self.device_port}'])
```

Comment:
if the host port passed in is already occupied, would the cmd fail and raise an error?

Reply:
Thanks for noticing a bug here. Fixed.

For most situations, it will raise an error. But if the port is occupied by adb forwarding, then adb will not raise an error but silently override the original binding. Added a check and raise an error for this situation.